chain/index/indexer.go

```diff
@@ -66,8 +71,13 @@ type SqliteIndexer struct {
 	updateSubs   map[uint64]*updateSub
 	subIdCounter uint64
 
+	started bool
+
 	closeLk sync.RWMutex
 	closed  bool
+
+	// ensures writes are serialized so backfilling does not race with index updates
+	writerLk sync.Mutex
```

Comment:
@rvagg Thoughts ? The purpose of this lock is to prevent races b/w real time updates and backfilling.

Reply:
do you think it's necessary? now we're only opening the db in a single process and we use transactions, shouldn't it be safe enough to allow multi-writes? particularly if they're idempotent or do you see some problems with the check-then-write process of updating the db?

Reply:
@rvagg 

1) Backfill gets a call for epoch e and reads ts at that epoch e
2) Real time indexing reverts epoch e
3) Backfill over writes ts at epoch e as applied

Reply:
ahh, I see the use in `ChainValidateIndex` where it's containing a complex logic set

yeah, this is reasonable, it's a cost but not huge, we can optimise later if we measure issues with it